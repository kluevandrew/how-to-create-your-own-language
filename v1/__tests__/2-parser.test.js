const { tokenize } = require('../1-tokenizer');
const { parse } = require('../2-parser');

describe('parse', () => {
  it('should parse', () => {
    expect(parse(tokenize('(print (a))'))).toMatchSnapshot();
  });
});
