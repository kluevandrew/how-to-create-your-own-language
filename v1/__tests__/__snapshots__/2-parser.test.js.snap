// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse should parse 1`] = `
Program {
  "expressions": Array [
    ParenExpression {
      "args": Array [
        ParenExpression {
          "args": Array [],
          "name": Identifier {
            "text": "a",
            "type": "Identifier",
          },
          "type": "Expression",
        },
      ],
      "name": Identifier {
        "text": "print",
        "type": "Identifier",
      },
      "type": "Expression",
    },
  ],
  "type": "Program",
}
`;
